examples/resnet-verify.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "glow/Base/Image.h"
+#include "glow/ExecutionEngine/ExecutionEngine.h"
+#include "glow/Graph/Hook.h"
+#include "glow/Importer/Caffe2ModelLoader.h"
+
+#include "llvm/Support/Format.h"
+
+using namespace glow;
+
+const char inputName[] = "gpu_0/data";
+
+class Tester {
+  Context ctx;
+  ExecutionEngine EE;
+  Module &mod;
+  Function *F;
+  TypeRef inputType;
+  Placeholder *input;
+  Placeholder *output;
+
+ public:
+  Tester(BackendKind kind)
+      : EE(kind),
+        mod(EE.getModule()),
+        F(mod.createFunction("resnet50")),
+        inputType(mod.uniqueType(ElemKind::FloatTy, {1, 3, 224, 224}))
+    {
+      // Load and compile ResNet-50.
+      Caffe2ModelLoader loader("resnet50/predict_net.pb",
+                               "resnet50/init_net.pb", {inputName},
+                               {inputType}, *F);
+      input = llvm::cast<glow::Placeholder>(loader.getNodeValueByName(inputName));
+      output = loader.getSingleOutput();
+    }
+
+  void bindInput(Tensor *batch) {
+    // Allocate memory for input and bind it to the placeholders.
+    ctx.allocate(mod.getPlaceholders());
+    updateInputPlaceholders(ctx, {input}, {batch});
+  }
+
+  TypeRef getInputType() const { return inputType; }
+
+  Function *getFunction() const { return F; }
+
+  Tensor *hookAndRun(llvm::StringRef name) {
```

Comment:
I have a small concern here. You try to hook by name, but names are not guaranteed to be unique in general. Should we also have a variant of `hookAndRun` that takes Node* as a parameter?

Reply:
Though, it is just an example. So, we probably don't care too much.

Reply:
Yes, this node-naming thing is a real problem.  You need to hook "the same" node in each network, but names aren't a strong guarantee.  It would be nice to be able to hook the function by node and then clone the function into the secondary backend, but cloning from one Module to another seemed kind of tricky.

Reply:
The hooking is at least taking place before backend-specific transformation takes place.  I'm not certain that the model loader always behaves the same regardless of backend, but it's better than e.g. after compile().

Reply:
I think that names are uniqe, except that they name may change due to compiler transformations.